engine/queue/score-queue-impl/src/main/java/io/cloudslang/engine/queue/repositories/ExecutionQueueRepositoryImpl.java

```diff
@@ -287,29 +348,123 @@ public int getBatchSize() {
         }
     }
 
+	@Override
+	public List<ExecutionMessage> poll(String workerId, int maxSize, long workerPollingMemory, ExecStatus... statuses) {
+
+		pollJdbcTemplate.setMaxRows(maxSize);
+		pollJdbcTemplate.setFetchSize(maxSize);
+
+		// prepare the sql statement
+		String sqlStat = QUERY_WORKER_WITH_PAYLOAD_SQL.replace(SKIP_LARGE_PAYLOADS_KEY, EMPTY);
+		sqlStat = sqlStat.replaceAll(":status", StringUtils.repeat("?", ",", statuses.length));
+
+		// prepare the arguments
+		List<Object> argsList = new LinkedList<>();
+		argsList.add(workerId);
+		for (ExecStatus status : statuses) {
+			argsList.add(status.getNumber());
+		}
+		argsList.add(workerPollingMemory);
+
+		List<ExecutionMessage> executionMessages = doSelectWithTemplate(
+						pollJdbcTemplate,
+						sqlStat,
+						new CustomRowMapperFactory().create(EXECUTION_MESSAGE),
+						argsList.toArray());
+
+		if (executionMessages.isEmpty()) {
+			if (logger.isDebugEnabled()) {
+				logger.debug("retry polling for " + workerId + " memory " + workerPollingMemory + " statuses " + statuses);
+			}
+
+			executionMessages = retryPollingSkippingLargePayloads(argsList, statuses, workerPollingMemory);
+		}
+
+		if (isMonitorLargeMessages()) {
+			executor.submit(() -> updateLargeMessages(workerId, workerPollingMemory));
+		}
+
+		return executionMessages;
+	}
+
+	private boolean isMonitorLargeMessages() {
```

Comment:
please remove isMonitorLargeMessage otherwise the job does not have records to process, since updateLargeMessage adds the records.